v2/app/src/main/java/com/google/firebase/example/makeitso/ui/home/HomeScreen.kt

```diff
@@ -1,14 +1,134 @@
 package com.google.firebase.example.makeitso.ui.home
 
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.rememberLazyListState
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Add
+import androidx.compose.material.icons.filled.Settings
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.ExtendedFloatingActionButton
+import androidx.compose.material3.Icon
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.material3.TopAppBarDefaults
+import androidx.compose.material3.rememberTopAppBarState
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import androidx.constraintlayout.compose.ConstraintLayout
+import androidx.constraintlayout.compose.Dimension
 import androidx.hilt.navigation.compose.hiltViewModel
 import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import com.google.firebase.example.makeitso.R
+import com.google.firebase.example.makeitso.data.model.TodoList
+import com.google.firebase.example.makeitso.ui.shared.CenterTopAppBar
+import com.google.firebase.example.makeitso.ui.shared.LoadingIndicator
+import com.google.firebase.example.makeitso.ui.theme.DarkBlue
+import com.google.firebase.example.makeitso.ui.theme.MediumYellow
 import kotlinx.serialization.Serializable
 
 @Serializable
 object HomeRoute
 
 @Composable
-fun HomeScreen(viewModel: HomeViewModel = hiltViewModel()) {
-    val uiState = viewModel.uiState.collectAsStateWithLifecycle()
+fun HomeScreen(
+    openSettingsScreen: () -> Unit,
+    viewModel: HomeViewModel = hiltViewModel()
+) {
+    val isLoadingUser by viewModel.isLoadingUser.collectAsStateWithLifecycle()
+
+    if (isLoadingUser) LoadingIndicator()
+    else HomeScreenContent(openSettingsScreen, viewModel)
```

Comment:
It's a [Kotlin convention](https://kotlinlang.org/docs/coding-conventions.html#control-flow-statements) to always use curly braces in a multiline `if` statement:
> If the condition of an if or when statement is multiline, always use curly braces around the body of the statement.

```suggestion
    if (isLoadingUser) {
        LoadingIndicator()
    } else {
        HomeScreenContent(openSettingsScreen, viewModel)
    }
```